api/src/services/notificationService.ts

```diff
@@ -20,4 +23,24 @@ export default class NotificationService {
 
         return notifications;
     }
+
+    public async readNotification(notificationId: number) {
+        const notification = await this.notificationRepository.findOne({
+            where: {
+                Id: notificationId,
+            },
+        });
+
+        if (!notification) {
+            throw new NotificationNotFoundError();
+        }
+
+        if (!ValidationUtils.isNull(notification.ReadAt)) {
+            throw new NotificationAlreadyReadError();
```

Comment:
Esse é um caso possível? visto que na sua query não inclui deleted e o readAt é um campo de deleted na entity certo?

Reply:
Acho que seria bom então adicionar na query o parâmetro se quiser retornar esse erro, senão acredito que só vai voltar not found

Reply:
Tem razão. Vou adicionar o withDeleted = true na query. Aí, pelo menos, se der erro, dá pra ter uma ideia melhor do que aconteceu.